### PR TITLE
Feature/pretrain refactor

### PR DESCRIPTION
**Pull request by wenshuoliu**
- add lr scheduler options
- save ckpts before finishing an epoch for pretrain-transformer-lm.py

### CODE REVIEW

File: api-examples/transformer_utils.py [hunk 1]
```diff
@@ -642,3 +643,37 @@ def create_model(embeddings, d_model, d_ff, dropout, num_heads, num_layers, mode
 
     logger.info(model)
     return model
+
+
+def save_checkpoint(model: torch.nn.Module, model_base: str, count: int, logger, tick_type: str = 'epoch'):
```
Comment (dpressel) [inline api-examples/transformer_utils.py:648]: This was already moved into 8mi
